src/pages/[platform]/build-a-backend/server-side-rendering/nuxt/index.mdx

```diff
@@ -0,0 +1,571 @@
+
+### Implement `01.amplify-apis.server.ts`
+
+Example implementation:
+
+<Accordion title='Expand to view the example implementation' headingLevel='4' eyebrow='Learn more'>
+
+```ts
+import {
+  createKeyValueStorageFromCookieStorageAdapter,
+  createUserPoolsTokenProvider,
+  createAWSCredentialsAndIdentityIdProvider,
+  runWithAmplifyServerContext
+} from 'aws-amplify/adapter-core';
+import { parseAmplifyConfig } from 'aws-amplify/utils';
+import {
+  fetchAuthSession,
+  fetchUserAttributes,
+  getCurrentUser
+} from 'aws-amplify/auth/server';
+import { list } from 'aws-amplify/storage/server';
+import { generateServerClient } from 'aws-amplify/api/server';
+import type { ListPaginateInput } from 'aws-amplify/storage';
+import type {
+  LibraryOptions,
+  FetchAuthSessionOptions
+} from '@aws-amplify/core';
+import type {
+  GraphQLOptionsV6,
+  GraphQLResponseV6
+} from '@aws-amplify/api-graphql';
+
+import config from '../amplifyconfiguration.json';
+
+type CookieRef = ReturnType<typeof useCookie<string | null | undefined>>;
+
+// parse the content of `amplifyconfiguration.json` into the shape of ResourceConfig
+const amplifyConfig = parseAmplifyConfig(config);
+
+// create the Amplify used token cookies names array
+const userPoolClientId = amplifyConfig.Auth!.Cognito.userPoolClientId;
+const lastAuthUserCookieName = `CognitoIdentityServiceProvider.${userPoolClientId}.LastAuthUser`;
+
+// create a GraphQL client that can be used in a server context
+const gqlServerClient = generateServerClient({ config: amplifyConfig });
+
+const getAmplifyAuthKeys = (lastAuthUser: string) =>
+  ['idToken', 'accessToken', 'refreshToken', 'clockDrift']
+    .map(
+      (key) =>
+        `CognitoIdentityServiceProvider.${userPoolClientId}.${lastAuthUser}.${key}`
+    )
+    .concat(lastAuthUserCookieName);
+
+// define the plugin
+export default defineNuxtPlugin({
+  name: 'AmplifyAPIs',
+  enforce: 'pre',
+  setup() {
+    // when setCookie is needed, set expirers as a year in this example
+    const expires = new Date();
+    expires.setDate(expires.getDate() + 365);
+
+    // get the last auth user cookie value
+    const lastAuthUserCookie = useCookie(lastAuthUserCookieName, {
+      sameSite: 'lax',
+      expires,
+      secure: true
+    });
+
+    // get all Amplify auth token cookie names
+    const authKeys = getAmplifyAuthKeys(lastAuthUserCookie.value!);
+
+    // create a key-value map of cookie name => cookie ref
+    const amplifyCookies = authKeys
+      .map((name) => ({
+        name,
+        cookieRef: useCookie(name, { sameSite: 'lax', expires, secure: true })
+      }))
+      .filter(
+        (item): item is { name: string; cookieRef: CookieRef } =>
+          item.cookieRef !== undefined
+      )
+      .reduce(
+        (result, current) => ({
+          ...result,
+          [current.name]: current.cookieRef
+        }),
+        {} as Record<string, CookieRef>
+      );
+
+    // create a key value storage based on the cookies
```

Comment:
If I understand correctly, this is a critical item, since an incorrect implementation, or reusing the storage in the wrong context, will cause cross-request pollution. Need more detail here to ensure customers know what's happening and why it's important

Reply:
Agreed. I will add the callout.